### ISSUE TITLE
Reference frames

### ISSUE THREAD

**Issue by kettenis** [casangi/xradio#339]
The list of possible reference frames includes the capitalized casacore reference frames, but I think some of those map directly onto the astropy ones (e.g. `'GEO'` is the same as `'gcrs'`)?

Is `'icrs'` indeed the most sensible choice for the default velocity reference frame?

The linked astropy documentation still has a warning that SpectralCoord is experimental (stating that it is new in Astropy v4.1). It is probably not wise to to base MSv4 on something that is expected to change at some point. Maybe clarification should be sought from the astropy developers about this?

**Comment by kettenis** [comment]
Also ITRF and ICRS have multiple realizations. The difference might be important for (geodetic) VLBI, although for current astrometric VLBI I don't think there is an expectation amongst users that things are labelled according to the actual realization of the reference frame that was used.

**Comment by FedeMPouzols** [comment]
Yes, I think that it was not clear whether the definitions are exactly the same. More than an objection I think that it was a lack of confidence that we can safely translate GEO=>GCRS (once done we'd forget about the hidden casacore "GEO").
There is also the caveat that GCRS in astropy has some flexibility and parameters that allow for example "This frame also includes frames that are defined relative to the Earth, but that are offset (in both position and velocity) from the Earth." which could also introduce ambiguity in the translation.

**Comment by FedeMPouzols** [comment]
I see two remaining points:
1. the capitalized casacore Spectral Coord  reference frames
1. The ICRS/ITRF frame realization

---
As for 1, the remaining capitalized casacore frames are "REST", "BARY", "GEO", and "TOPO". "LSRK" and "LSRD" are already translated to "lsrk" and "lsrd". I believe that "REST", "TOPO", and "BARY" have to remain as not having a direct equivalent for now. With "GEO" we could:
- A) translate it to "gcrs" (which is probably fine but who knows) and add in the schema docstrings or somewhere in the detailed documentation of the MSv2 converter that GEO=>gcrs, LSRK=>lsdk, LSRD=>lsrd to make users aware of possible hidden issues.
- B) Keep it capitalized, in case.

I think we can take the (unknown?) risk and do A), please let me know if there is any objections.

---
As for 2, would a  `frame_realization` field in the SpectralCoord dicts satisfy the use case of geodecit VLBI? Is this currently being used? The converter would currently set it to "Unknown" in any case.

---
I've put some preliminary changes in the branch of this issue and opened this draft PR: #371. This is for review purposes, it could be merged there or it could be merged into other ongoing work/branches (note to self: the FreqSamplingArray change would need to be sorted out if merging into 350).
